File: app/src/main/java/com/battlelancer/seriesguide/extensions/VodsterExtension.java
package com.battlelancer.seriesguide.extensions;

import android.content.Intent;
import android.net.Uri;
import com.battlelancer.seriesguide.R;
import com.battlelancer.seriesguide.api.Action;
import com.battlelancer.seriesguide.api.Episode;
import com.battlelancer.seriesguide.api.Movie;
import com.battlelancer.seriesguide.api.SeriesGuideExtension;

/**
 * Searches vodster.de for shows and movies.
 */
public class VodsterExtension extends SeriesGuideExtension {

    public static final String VODSTER_SEARCH_URL = "http://www.vodster.de?";

    public VodsterExtension() {
        super("VodsterExtension");
    }

    @Override
    protected void onRequest(int episodeIdentifier, Episode episode) {
        publishVodsterAction(episodeIdentifier, "tvdb=" + episode.getShowTvdbId());
    }

    @Override
    protected void onRequest(int movieIdentifier, Movie movie) {
        publishVodsterAction(movieIdentifier, "tmdb=" + movie.getTmdbId());
    }

    private void publishVodsterAction(int identifier, String query) {
        String uri = VODSTER_SEARCH_URL + query;
        publishAction(new Action.Builder(getString(R.string.extension_vodster), identifier)
                .viewIntent(new Intent(Intent.ACTION_VIEW).setData(Uri.parse(uri)))
                .build());
    }
}

File: app/src/main/java/com/battlelancer/seriesguide/extensions/VodsterExtension.kt
package com.battlelancer.seriesguide.extensions

import android.content.Intent
import android.net.Uri
import com.battlelancer.seriesguide.R
import com.battlelancer.seriesguide.api.Action
import com.battlelancer.seriesguide.api.Episode
import com.battlelancer.seriesguide.api.Movie
import com.battlelancer.seriesguide.api.SeriesGuideExtension

/**
 * Searches vodster.de for shows and movies.
 */
class VodsterExtension : SeriesGuideExtension("VodsterExtension") {

    override fun onRequest(episodeIdentifier: Int, episode: Episode) {
        val showTvdbId = episode.showTvdbId
        if (showTvdbId != 0) {
            publishVodsterAction(episodeIdentifier, "tvdb=$showTvdbId")
        }
    }

    override fun onRequest(movieIdentifier: Int, movie: Movie) {
        publishVodsterAction(movieIdentifier, "tmdb=${movie.tmdbId}")
    }

    private fun publishVodsterAction(identifier: Int, query: String) {
        val uri = VODSTER_SEARCH_URL + query
        publishAction(
            Action.Builder(getString(R.string.extension_vodster), identifier)
                .viewIntent(Intent(Intent.ACTION_VIEW).setData(Uri.parse(uri)))
                .build()
        )
    }

    companion object {
        private const val VODSTER_SEARCH_URL = "https://www.vodster.de?"
    }
}